x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/action_connector_form/create_connector_flyout/index.tsx

```diff
@@ -215,6 +218,46 @@ const CreateConnectorFlyoutComponent: React.FC<CreateConnectorFlyoutProps> = ({
     };
   }, []);
 
+  const banner = useMemo(() => {
+    const redirectToInferenceConnector = (provider: string) => {
+      if (allActionTypes) {
+        setActionType(allActionTypes['.inference']);
+
+        const newSearchParams = new URLSearchParams(searchParams);
+        newSearchParams.set('connectorProvider', provider);
+        // Update the URL without a full page refresh
+        setSearchParams(newSearchParams);
+      }
+    };
+
+    const banners = [];
+    if (!actionType && hasActionsUpgradeableByTrial) {
+      banners.push(<UpgradeLicenseCallOut />);
+    }
+
+    if (actionTypeModel?.id && actionTypeModel?.isDeprecated && actionTypeModel?.actionTypeTitle) {
+      banners.push(
+        banners.length ? <EuiSpacer /> : null,
+        <DeprecatedAIConnectorCallOut
+          name={actionTypeModel.actionTypeTitle}
+          id={actionTypeModel.id}
+          onClick={redirectToInferenceConnector}
+        />
+      );
+    }
```

Comment:
based on my suggestions above, all this code (except the UpgradeLicenseCallOut) would live live in the `deperecationComponent` and be generic for all connector types.